### PR TITLE
IAMA-3806 Add aggregate error package

### PR DESCRIPTION
**Pull request by GrzZebrowski**
old lib files removed, selvera-aggregate-error package used instead

### CODE REVIEW

File: test/transaction.spec.ts [hunk 1]
```diff
@@ -145,7 +145,9 @@ describe('Transaction wrapper', () => {
             if (!(error instanceof AggregateError)) {
                 done.fail(new Error('Error should be an aggregate error, instead got: ' + typeof error));
             } else {
-                expect(error.errors.length).toBe(2);
+                expect(error.cause).toBeDefined();
+                expect(error.message).toEqual('this.client.release is not a function');
+                expect(error.cause!.message).toEqual('Failure!');
```
Comment (GrzZebrowski) [inline test/transaction.spec.ts]: Please check if that relation is ok

File: lib/transaction.ts [hunk 2]
```diff
@@ -42,12 +42,13 @@ export class Transaction {
             await this.client.query('COMMIT');
             this.client.release();
         } catch (error) {
+            const commitError = new AggregateError(error);
```
Comment (pcwiek) [inline lib/transaction.ts]: Techically we don't have to wrap the source error. We can just do it on line 49

File: lib/transaction.ts [hunk 3]
```diff
@@ -68,12 +69,13 @@ export class Transaction {
             await this.client.query('ROLLBACK');
             this.client.release();
         } catch (error) {
+            const rollbackError = new AggregateError(error);
```
Comment (pcwiek) [inline lib/transaction.ts]: Techically we don't have to wrap the source error. We can just do it on line 76

File: test/transaction.spec.ts (hunk 1 above)
Comment (pcwiek) [reply test/transaction.spec.ts]: I think it looks about right

File: lib/transaction.ts (hunk 2 above)
Comment (GrzZebrowski) [reply lib/transaction.ts]: In that case we would have duplication in line 49 and 51 of wrapping the same error, but it's cosmetic matter. I'll make a suggested change.

Comment (pcwiek) [reply lib/transaction.ts]: @GrzZebrowski nah, I just mean that if there is no 'second' error, we can rethrow the original one: `throw error;`
If we get the 'inner' exception, then we can do `throw new AggregateError(inner, error);`

Comment (GrzZebrowski) [reply lib/transaction.ts]: Ah, ok, sounds good.

File: test/transaction.spec.ts [hunk 4]
```diff
@@ -130,7 +130,7 @@ describe('Transaction wrapper', () => {
         const failingClient: any = {
             query(text: string) {
                 if (text === 'COMMIT' || text === 'ROLLBACK') {
-                    return Promise.reject('Failure!');
```
Comment (GrzZebrowski) [inline test/transaction.spec.ts]: TypeError: Expected the cause to be an Error, instead got: string
    at AggregateError (C:\Dev\slvr\vagrant-local\nodes\npm-pg\node_modules\selvera-aggregate-error\dist\lib\aggregate-error.js:32:19)
Do we assume such case is irrelevant?

Comment (pcwiek) [reply test/transaction.spec.ts]: @GrzZebrowski yeah it should be an `Error` instance 😉

**Pull request merged**